### PR TITLE
Integrate OCC nurbs through pluggables

### PR DESCRIPTION
**Pull request by tomvanmele**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [x] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by tomvanmele** [comment]
have separated this from the Rhino interop PR since they are independent contributions...

File: src/compas/geometry/curves/nurbs.py [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+from compas.geometry import Point
+from compas.geometry import Frame
+from compas.utilities import linspace
+
+from .curve import Curve
+
+
+@pluggable(category='factories')
+def new_nurbscurve(cls, *args, **kwargs):
+    raise NotImplementedError
```
Comment (gonzalocasas) [inline src/compas/geometry/curves/nurbs.py:17]: I will keep bugging about this: a specific exception that clearly states this is an extension-point that is filled in by code somewhere else would be very good. A lot of our users read the source code, and this leads them to think the feature is literally not implemented, but that is not really this case for it.

**Review by gonzalocasas — APPROVED** [review]
LGTM

File: src/compas/geometry/curves/nurbs.py (hunk 1 above)
Comment (tomvanmele) [reply src/compas/geometry/curves/nurbs.py:17]: will do this in separate pr for all pluggables

**Pull request merged**